src/PollySandbox/ResilienceStrategyFactory.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Martin Costello, 2023. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+
+using Polly;
+using Polly.CircuitBreaker;
+using Polly.Fallback;
+using Polly.Timeout;
+using Refit;
+
+namespace PollySandbox;
+
+public partial class ResilienceStrategyFactory
+{
+    private readonly ResilienceStrategyStore _store;
+    private readonly IMetricsPublisher _metricsPublisher;
```

Comment:
Is `IMetricsPublisher` even  necessary anymore with V8?

Reply:
It is without also doing additional engineering work to hook into wherever Polly v8 is sending these metrics to and then re-publishing them to where our metrics go to in our production systems.

Reply:
You can also hook into Polly metering in universal way and just forward it to `IMetricsPublisher` (if you don't want to consume native Polly metrics).


``` csharp
public static void ForwardPollyMetering()
{
    var meterListener = new MeterListener
    {
        InstrumentPublished = (instrument, listener) =>
        {
            if (instrument.Meter.Name == "Polly" && instrument.Name == "resilience-events")
            {
                listener.EnableMeasurementEvents(instrument);
            }
        }
    };
    meterListener.SetMeasurementEventCallback<int>(OnMeasurementRecorded);
    meterListener.SetMeasurementEventCallback<double>(OnMeasurementRecorded);
    meterListener.Start();

    void OnMeasurementRecorded<T>(Instrument instrument, T measurement, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
    {
        var eventName = $"{GetValue(tags, "strategy-type")}-{GetValue(tags, "event-name")}-{GetValue(tags, "operation-key")}";
    }

    static object GetValue(ReadOnlySpan<KeyValuePair<string, object?>> tags, string key)
    {
        foreach (var pair in tags)
        {
            if (pair.Key == key)
            {
                return pair.Value ?? "Unknown";
            }
        }

        return "Unknown";
    }
}

```